exporter/sapmexporter/exporter.go

```diff
@@ -126,17 +125,25 @@ func (se *sapmExporter) pushTraceData(ctx context.Context, td ptrace.Traces) err
 	return nil
 }
 
-func (se *sapmExporter) retrieveAccessToken(md ptrace.ResourceSpans) string {
+func (se *sapmExporter) retrieveAccessToken(ctx context.Context, md ptrace.ResourceSpans) string {
 	if !se.config.AccessTokenPassthrough {
 		// Nothing to do if token is pass through not configured or resource is nil.
 		return ""
 	}
 
-	attrs := md.Resource().Attributes()
-	if accessToken, ok := attrs.Get(splunk.SFxAccessTokenLabel); ok {
-		return accessToken.Str()
+	var token string
+	cl := client.FromContext(ctx)
+	ss := cl.Metadata.Get(splunk.SFxAccessTokenHeader)
+	if len(ss) > 0 {
+		token = ss[0]
+	} else {
+		attrs := md.Resource().Attributes()
+		if accessToken, ok := attrs.Get(splunk.SFxAccessTokenLabel); ok {
+			token = accessToken.Str()
+		}
```

Comment:
just to simplify and keep the changeset cleaner
```suggestion
	if len(ss) > 0 {
		return ss[0]
	}
	attrs := md.Resource().Attributes()
	if accessToken, ok := attrs.Get(splunk.SFxAccessTokenLabel); ok {
		return accessToken.Str()
	}
	return ""
```